File: python/queue/lib/Queue.py
# FIFO: first in, first out

class Queue:

  def __init__(self):
    self.items = []

  def clear(self):
    self.items = []

  def size(self):
    return len(self.items)

  def add(self, item):
    self.items.append(item)

  def remove(self):
    return self.items.pop(0)

  def front(self):
    return self.items[0]

  def is_empty(self):
    return self.size() == 0

File: ruby/queue/queue.rb
# FIFO: first in first out

class Queue

  #attr_accessor :items
  #attr_writer :items
  attr_reader :items

  def initialize
    @items = []
  end

  def add(item)
    @items.push(item)
  end

  def remove
    @items.shift()
  end

  def front
    @items.first()
  end

  def is_empty
    @items.length() == 0
  end

  def size
    @items.length()
  end

  def clear
    @items = []
  end
end

